From 0115b02f43d23612366d0c5063e6b314d04e3853 Mon Sep 17 00:00:00 2001
From: Klaus von der Heyde <39677540+asc-soc@users.noreply.github.com>
Date: Mon, 16 Sep 2019 19:23:00 +0200
Subject: [PATCH]  Added special handling when Midori.Urlbar is in focus (#366)

If the urlbar has focus and reacts on the event, stop further handling. This enables ^N and ^T again, as the edit field does not handle them. Any key binding in the edit field takes precedence.

Fixes: #364
---
 core/browser.vala | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/browser.vala b/core/browser.vala
index 3ce21e40..503aa301 100644
--- a/core/browser.vala
+++ b/core/browser.vala
@@ -453,6 +453,12 @@ namespace Midori {
             // No keyboard shortcuts in locked state
             if (is_locked) {
                 return propagate_key_event (event);
+            }
+			// Default behaviour for navigation bar
+            if (get_focus () is Midori.Urlbar) {
+                if (navigationbar.urlbar.key_press_event (event)) {
+                    return true;
+                }
             }
             // Default behavior for standard widgets
             if (!(get_focus () is WebKit.WebViewBase)) {
